### ISSUE TITLE
Help: PrismaClient Returning Undefined using Fastify

### ISSUE THREAD

**Issue by MarioWork** [fastify/help#699]
## 💬  Question here 

Hello there, I have a project where I am using the Web Framework Fastify with Prisma and my ´PrismaClient´ is always returning undefined. I have been trying to fix this problem but cannot seem to know how, the connection works fine but the ´PrismaClient´is not working. 
I started the project with `npm init fastify´
Thank you in advance.

 [github repo ](https://github.com/MarioWork/mcu-api/tree/mario/feat/generate-api-key-route)

## Your Environment

- *node version*: 18.2.0
- *fastify version*: 3.0.0
- *os*: MacOs

**Comment by kibertoad** [comment]
don't forget to call await app.ready if you are not awaiting registration itself

**Comment by MarioWork** [comment]
> don't forget to call await app.ready if you are not awaiting registration itself

It is registering right, because it does not give any error when connecting to the database but the `new PrismaClient()`is always returning `undefined`

**Comment by kibertoad** [comment]
how is using constructor related to fastify usage?

**Comment by MarioWork** [comment]
> how is using constructor related to fastify usage?

Because if I do not use fastify everything works, but when I use fastify and put the `prisma initialization` into a plugin it does...

**Comment by Eomm** [comment]
The `.to()` method is not a fastify stuff.
Who adds it?

```
const [error] = await fastify.to(prisma.$connect());
```

> always returning undefined

Add a complete stacktrace would be more helpful

**Comment by kibertoad** [comment]
https://www.prisma.io/fastify

I don't see anything special in their plugin example, but you can try copying it verbatim.

which part of the code throws now? is it literally after you invoke constructor? is that part of the code invoked at all?

**Comment by MarioWork** [comment]
> 
[fastify-sensible](https://github.com/fastify/fastify-sensible)

**Comment by MarioWork** [comment]
> https://www.prisma.io/fastify

It does not throw an exception... It simply returns `undefined` and so I can not use the instance because its `undefined`

**Comment by kibertoad** [comment]
what returns undefined? constructor can't do that. please add some logging to understand what is happening

**Comment by MarioWork** [comment]
> 

If I log `new PrismaClient()`it logs `undefined`that is how I found the problem

**Comment by MarioWork** [comment]
`fastify.log.info(new PrismaClient());` logs `undefined`

**Comment by kibertoad** [comment]
you are logging constructor, not created instance in repro

**Comment by MarioWork** [comment]
> logging
Yes I did not push mb, but even if you log as I sent before it logs `undefined`

**Comment by kibertoad** [comment]
please push full repro

**Comment by MarioWork** [comment]
> please push full repro

Just did

**Comment by kibertoad** [comment]
thank you, will take a look in few hours when by computer

**Comment by MarioWork** [comment]
> thank you, will take a look in few hours when by computer

Thank you very much, I have been stuck in this problem for a long time... That is why I reached 👍🏻

**Comment by Eomm** [comment]
Following the docs, this should works:

```
const register = async (fastify) => {
    fastify.log.info(`Registering ${metadata.name} plugin`);

    const prisma = new PrismaClient();

    try {
      await prisma.$connect()
    } catch (error) {
        fastify.log.error('Failed to connect to database: \n' + error);
        throw error;
    }

    fastify.decorate('prisma', prisma);
}
module.exports = fp(register, metadata);
```

**Comment by MarioWork** [comment]
> 

Yes but it is not working for me, and I cannot understand why... and it only does not work with fastify...

**Comment by mcollina** [comment]
I would actually recommend to open an issue on the prisma community. They have significantly better means than us to to help.

**Comment by MarioWork** [comment]
> I would actually recommend to open an issue on the prisma community. They have significantly better means than us to to help.

I guess... but since it is not working with fastify, I tought to create in this repo.

**Comment by mcollina** [comment]
I've tried this and everything works as expected (there were a few conflicts committed to the repo).

**Comment by mcollina** [comment]
master

**Comment by MarioWork** [comment]
> master

I do not understand what I am doing wrong, but it is always giving me undefined :/
